### DISCUSSION TITLE
When does compatibility flag become available for production use?

### DISCUSSION THREAD

**Discussion by danbars** [cloudflare/workers-sdk#2858] [Archive]
Following [this merged PR](https://github.com/cloudflare/workers-sdk/pull/2539/files), I've tried using `compatibility_flags['nodejs_compat']` in my `wrangler.toml`, so I can then `import { AsyncLocalStorage } from 'node:async_hooks'` in my code.

I was happy to see that build have passed.
However, I'm still getting this:
```
...
 text: 'A request to the Cloudflare API
  (/accounts/c4a44021bc47f9a2fa9f08bf99d81710/workers/scripts/submission-handler/edge-preview)
  failed.',
    notes: [
      {
        text: 'The compatibility flag nodejs_compat is experimental and cannot yet be used in
  Workers deployed to Cloudflare.\n' +
          ' [code: 10021]'
      }
    ],
    location: undefined,
    kind: 'error',
    code: 10021
  }
```

I'm using latest wrangler v2.12.0 and node v18.12.1

My question is what is the lifecycle of experimental flags? 
When can I expect it to become available for use in production?

Thanks

**Comment by JacobMGEvans** [comment]
Hey thanks for the question!! 

The flag was just recently moved from internal experimental to publicly accessible. If you are still having the issue please feel free to report the issue and directly ping me in it 😄

**Comment by danbars** [comment]
Tried it and it works!

Thanks 👍
